src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
>How so? ... QPC has a resolution of 1 microsecond or better. A 52-bit mantissa can represent 2^52 microseconds without losing precision.

This is not true. The LSB of the value returned by QPC does not represent 1 microsecond. As I said above, QPC is allowed to roll over after 100 years (2^32 seconds) so it means QPC is allowed to use 32 of the 64 bits for one second.

Indeed, on the machine I have right now, two QPCs run about 1 second apart differ by 22 bits.

>Keep in mind, that what matters is not the absolute value returned by QPC.

The current code converts the absolute value returned by QPC to a double, so yes the absolute value matters.